cockpit/gui/imageViewer/viewCanvas.py

```diff
@@ -703,12 +704,19 @@ def onMouse(self, event):
             self.mouseDragY = self.curMouseY
         elif event.RightDown():
             # Show a menu.
-            menu = wx.Menu()
-            for label, action in self.getMenuActions():
-                id = wx.NewIdRef()
-                menu.Append(id, label)
-                self.Bind(wx.EVT_MENU,  lambda event, action = action: action(), id= id)
-            cockpit.gui.guiUtils.placeMenuAtMouse(self, menu)
+            if self._menu is None:
```

Comment:
Can the menu be constructed when the canvas is constructed? This would save a check for its existence each time right click is performed. It would also make things simpler if we ever add a method to get to this menu without a right click.

Reply:
Maybe. It references methods on image, so I need to check that's created in __init__.

Reply:
Resolved in 41d02051f798ed830d4337dfe60b3961df64ac1c